crates/field/src/scalars_collection.rs

```diff
@@ -0,0 +1,366 @@
+// Copyright 2025 Irreducible Inc.
+
+use crate::{
+	packed::{get_packed_slice_unchecked, len_packed_slice, set_packed_slice_unchecked},
+	PackedField,
+};
+
+/// A trait for collections of scalars, allowing for indexed access.
+/// This trait is used to abstract over different types of collections - scalar slices,
+/// slices of packed field elements including subranges of collections.
+pub trait ScalarsCollection<F> {
```

Comment:
I think this should actually go in `binius_util` -- the trait definition isn't specific to fields. I'd call it something more generic like

```rust
pub trait RandomAccessSequence<T: Copy>` {}
```

Reply:
Applied